src/skeptic/lib.rs

```diff
@@ -784,3 +797,131 @@ fn test_sanitization_of_testnames() {
+        "\n".to_owned(),
+        "    panic!(\"I should panic\");\n".to_owned(),
+        "\n".to_owned(),
+        "}\n".to_owned(),
+        "```\n".to_owned(),
+        "\n".to_owned(),
+        "Rust code that should panic when compiling it.\n".to_owned(),
+        "\n".to_owned(),
+        "```rust,no_run,should_panic\n".to_owned(),
+        "\n".to_owned(),
+        "fn add(a: u32, b: u32) -> u32 {\n".to_owned(),
+        "\n".to_owned(),
+        "    a + b\n".to_owned(),
+        "\n".to_owned(),
+        "}\n".to_owned(),
+        "\n".to_owned(),
+        "\n".to_owned(),
+        "\n".to_owned(),
+        "\n".to_owned(),
+        "\n".to_owned(),
+        "fn main() {\n".to_owned(),//33
+        "\n".to_owned(),
+        "    add(1);\n".to_owned(),
+        "\n".to_owned(),
+        "}\n".to_owned(),
+        "```\n".to_owned(),
+
+    ];
+    let mut old_template = None;
+
+    let tests = extract_tests_from_string(&mut create_test_input(lines), &String::from("blah"), &mut old_template);
+
+    let test_start_line_numbers: Vec<String> = tests.into_iter().map(|test| get_line_number_from_test_name(test)).collect();
+
+    assert_eq!(test_start_line_numbers, vec!("11", "33"));
+}
+
+#[test]
+fn test_extracting_tests_from_string_states_the_line_number_as_the_beginning_of_the_code_block_but_this_time_for_sections() {
+    let lines = &[
+        "## Test Case  Names   With    weird     spacing       are        generated      without        error.\n".to_owned(),
+        "\n".to_owned(),
+        "```rust\n".to_owned(),
+        "struct Person<'a>(&'a str);\n".to_owned(),
+        "fn main() {\n".to_owned(),//5
+        "  let _ = Person(\"bors\");\n".to_owned(),
+        "}\n".to_owned(),
+        "```\n".to_owned(),
+        "\n".to_owned(),
+        "## !@#$ Test Cases )(() with {}[] non alphanumeric characters ^$23 characters are \"`#`\" generated correctly @#$@#$  22.\n".to_owned(),
+        "\n".to_owned(),
+        "```rust\n".to_owned(),
+        "struct Person<'a>(&'a str);\n".to_owned(),
+        "fn main() {\n".to_owned(),//14
+        "  let _ = Person(\"bors\");\n".to_owned(),
+        "}\n".to_owned(),
+        "```\n".to_owned(),
+        "\n".to_owned(),
+        "## Test cases with non ASCII ö_老虎_é characters are generated correctly.\n".to_owned(),
+        "\n".to_owned(),
+        "```rust\n".to_owned(),
+        "struct Person<'a>(&'a str);\n".to_owned(),
+        "fn main() {\n".to_owned(),//23
+        "  let _ = Person(\"bors\");\n".to_owned(),
+        "}\n".to_owned(),
+        "```\n".to_owned(),
+    ];
+    let mut old_template = None;
+
+    let tests = extract_tests_from_string(&mut create_test_input(lines), &String::from("blah"), &mut old_template);
+
+    let test_start_line_numbers: Vec<String> = tests.into_iter().map(|test| get_line_number_from_test_name(test)).collect();
+
+    assert_eq!(test_start_line_numbers, vec!("5", "14", "23"));
+}
+
+fn get_line_number_from_test_name(test: Test) -> String {
```

Comment:
this function generates unused warning it can be avoided by moving it along with tests into a separate module "tests" cfg'ed for test build.